base64.go

```diff
@@ -0,0 +1,29 @@
+package matrixfederation
+
+import (
+	"encoding/base64"
+	"encoding/json"
+)
+
+// A Base64String are some bytes encoded using base64 in JSON.
+type Base64String []byte
+
+// MarshalJSON encodes the bytes as base64 and then encodes the base64 as a JSON string.
+func (b64 Base64String) MarshalJSON() ([]byte, error) {
+	// This could be made more efficient by using base64.RawStdEncoding.Encode
+	// to write the base64 directly to the JSON. We don't need to JSON escape
+	// any of the characters used in base64.
+	return json.Marshal(base64.RawStdEncoding.EncodeToString(b64))
+}
+
+// UnmarshalJSON decodes a JSON string and then decodes the resulting base64.
+func (b64 *Base64String) UnmarshalJSON(raw []byte) (err error) {
```

Comment:
Here you're using a pointer receiver but `MarshalJSON` uses a value receiver. Intentional?